src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat.pop());
+  const dateObj = {};
+
+  fromFormat.forEach(function (part, i) {
+    dateObj[part] = dateParts[i];
+  });
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    return year.length === 4 ? year.slice(2) : year;
```

Comment:
The `convertYear` function does not handle the case where the year part is not present in `dateObj`. This could lead to accessing `undefined` values and result in incorrect conversions. Since we assume that the input is always correct, this might not be an issue, but it's something to keep in mind if the function is used in a different context.